rustls/src/client/hs.rs

```diff
@@ -419,6 +376,56 @@ fn emit_client_hello_for_retry(
     }
 }
 
+fn prepare_resumption<'a>(
```

Comment:
It was the case before you refactored as well, but I think this function is a bit light on test coverage for some of the branching points.

My runs are showing no coverage for:

* The `return None` when `!config.enable_tickets`
* The `None` and `_` branches of the `let resuming = match resuming` statement
* The two `Some` branches of the `match suite` check
* The `Some` branch that calls `can_resume_from` on `suite`

Give some of the confusion around the `enable_tickets` bool it would be nice to shore  up coverage while refactoring if isn't too gnarly. WDYT?

Reply:
Did you look at bogo coverage as well? It sure seemed to be able to catch a number of minor variants of this code as I was refactoring.

Reply:
Ah sorry! I didn't.

Is there a trick to view the generated coverage emitted by those runs? I've been using my IDE's ability to run the Cargo test suite with coverage collection and I'm not sure I can do the same through the Bogo runner :thinking: Do you have an existing workflow that's good for that?

Reply:
No, I usually look at the CodeCov stuff posted in PRs but apparently Codecov has been having issues lately.

Reply:
I can't seem to get CodeCov to re-run the failed task, but did find [this output](https://github.com/rustls/rustls/pull/1239/checks?check_run_id=12249465545) that indicates the `if !config.enable_tickets` return is missing coverage, as well as the `!config.supports_version(ProtocolVersion::TLSv1_3)` return. 

Installing `cargo llvm-cov` I was able to get generated `profraw` files with `LLVM_COV=$(which llvm-cov) LLVM_PROFDATA=$(which llvm-profdata) ./admin/coverage` but there's a failure to merge the results ("Unsupported instrumentation profile format version"). I might tilt at that more if I have some time.

In either event, I don't think this coverage is important enough to hold up this work.It's clearly an improvement to me :+1: